### ISSUE TITLE
Embedded workflows not loading in ComfyUI

### ISSUE THREAD

**Issue by lingondricka2** [Chaoses-Ib/ComfyScript#67]
When loading a png with embedded workflow into ComfyUI:

If a node is missing the workflow is not loaded at all
If all custom nodes are installed the workflow is loaded but all nodes are stacked ontop of each other with no connections.

This only apply to png's generated using ComfyScript, I've tried loading png's generated over 6 months ago also.

edit: Tested on standalone ComfyUI installation from last year, same result

**Comment by Chaoses-Ib** [comment]
It works on my side with latest ComfyUI commit https://github.com/comfyanonymous/ComfyUI/commit/5cbaa9e07c97296b536f240688f5a19300ecf30d:

![image](https://github.com/user-attachments/assets/ab04d5bc-5803-499c-835f-c362ba292c17)

An old image generated at 2023-12-26 also works.

Maybe it's some custom node messed it up? The only difference between the workflow metadata generated by ComfyScript and ComfyUI is ComfyScript uses string node ids, but that shouldn't be a problem.

**Comment by lingondricka2** [comment]
Same results loading the png online using https://www.runcomfy.com/comfyui-web, this is a serious head scratcher

**Comment by Chaoses-Ib** [comment]
That's really weird, it works on my side too. Have you tried another browser?

![image](https://github.com/user-attachments/assets/c3a1e0a8-6be8-410d-a008-9f2b8fe00169)

![image](https://github.com/user-attachments/assets/47f67bd3-7904-412e-b1c3-2a18618fbd1a)

**Comment by Chaoses-Ib** [comment]
Here is the image generated on my side:

![ComfyUI_00036_](https://github.com/user-attachments/assets/ec485053-88f7-4d3d-8d75-3881477050d2)

If this works, then the bug should be caused by some configuration difference.

**Comment by lingondricka2** [comment]
Yeah, chrome and firefox.

The image you provided is loaded, but the nodes are stacked ontop of each other with no connections.

**Comment by Chaoses-Ib** [comment]
Ok, now I'm getting your result with the same ComfyUI version, args and nodes. What a nasty bug... Though https://www.runcomfy.com/comfyui-web still works for me.

![image](https://github.com/user-attachments/assets/c1fe2630-28fc-4e09-8447-7af772f46661)

**Comment by Chaoses-Ib** [comment]
The support for non-integer node id is is broken in the new frontend: https://github.com/Comfy-Org/ComfyUI_frontend/issues/741

A temp workaround is to use the legacy frontend:
```sh
python main.py --front-end-version Comfy-Org/ComfyUI_legacy_frontend@latest
```

**Comment by lingondricka2** [comment]
Thank you, that fixes that problem for me, though the problem with workflow not loading if a node pack is missing is still there, but not really a issue for me since it's only for the convenience of using the manager to install the missing nodes,  I'm happy with this.

**Comment by Chaoses-Ib** [comment]
That problem applies to all images not generated by the web UI. Dropping an API format workflow JSON with missing node has the same problem:

![image](https://github.com/user-attachments/assets/dd85b452-bdeb-49a9-bc5c-4d171db0a809)

That being said, there is actually already a converter for converting API format to web UI format in the transpiler. So adding web UI workflow to the generated image is not hard. However, doing this may break some custom nodes. For example, if it has some hacks in the web UI to inject some info into the workflow, then the converted workflow will not have them, and then the custom node will break the generation process now. That's why I didn't add them before.

If you want, you can try to make a feature request in the frontend repository and see if they are willing to solve this.

**Comment by Chaoses-Ib** [comment]
The non-integer node id support is fixed in the latest frontend version v1.2.47. It can be manually installed by:
```sh
python main.py --front-end-version Comfy-Org/ComfyUI_frontend@latest
```
Or just wait a week for the default version used by ComfyUI to be updated.
